### DISCUSSION TITLE
Can i update custom_edit_url in loader

### DISCUSSION THREAD

**Discussion by laojun** [facebook/docusaurus#8786] [Q&A] [answered]
Hi, 
I have a lot of md files, And i don't want to add `custom_edit_url` one by one, So i try to use plugin to add a md loader, to update the `custom_edit_url`, But it doesn't seem to work

Here is my code
```
configureWebpack(config){
   return {
       module: {
          rules: [
            {
              test: /\.md$/,
              use: [{ loader: resolve('./set-md-custom-edit-url.ts') }],
            },
          ],
        },
   }
}
```

set-md-custom-edit-url.ts
```
module.exports = function (source, x, y, z) {
  let curPath = this.resourcePath?.replace(this.rootContext, '');
  if (curPath.includes('${{custom_edit_url}}:')) {
   console.log('Before', source);
   const newSource = source.replace(
      '${{custom_edit_url}}:',
      `custom_edit_url: https://github.com/my-pro/test/blob/develop${curPath}`
    );
   console.log('After', newSource)
    return newSource
  }
  return source;
};
```

The file updated well, But there is no `Edit this page` when i visit

Before
![image](https://user-images.githubusercontent.com/15364706/225882728-702196cd-5f68-4b5a-ad87-c46039250fb4.png)

After
![image](https://user-images.githubusercontent.com/15364706/225882900-4784b365-4ca9-420f-a873-048814d3243b.png)

Thank you!

**Comment by slorber** [comment, marked as answer]
The docs mention that the `editUrl` docs/blog config can be a function: this allows you to plug your own logic for computing the final edit url for each doc or blog post.

https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#EditUrlFunction

https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-blog#EditUrlFn

```tsx
        editUrl: ({versionDocsDirPath, docPath}) =>
          `https://github.com/facebook/docusaurus/edit/main/website/${versionDocsDirPath}/${docPath}`,
```

---

We also plan to introduce a `createFrontMatter` API later: https://github.com/facebook/docusaurus/issues/5568 => it could also allow you to insert a custom_edit_url to all docs dynamically

**Reply by laojun** [reply]
Thank you for the reply
